packer_test/common/check/file_gadgets.go

```diff
@@ -33,3 +34,32 @@ func FileExists(filePath string, isDir bool) Checker {
 		isDir:    isDir,
 	}
 }
+
+type fileInDir struct {
+	filename string
+	dirPath  string
+}
+
+func (fe fileInDir) Check(_, _ string, _ error) error {
+	files, err := os.ReadDir(fe.dirPath)
+	if err != nil {
+		return fmt.Errorf("failed to read dir %q: %s", fe.dirPath, err)
+	}
+
+	pattern := regexp.MustCompile(fe.filename)
```

Comment:
If the idea is to glob files, you can probably use `filepath.Glob` here instead, that'd simplify the whole check